### PR TITLE
#720 - Adding support for DateInterval deserialization

### PR DESCRIPTION
**Pull request by c0ntax**
Added in DateInterval deserialization and associated unit test

Fixes https://github.com/schmittjoh/serializer/issues/720

### CODE REVIEW

File: src/JMS/Serializer/Handler/DateHandler.php [hunk 1]
```diff
@@ -173,6 +191,18 @@ private function parseDateTime($data, array $type, $immutable = false)
         return $datetime;
     }
 
+    private function parseDateInterval($data)
+    {
+        $dateInterval = null;
+        try {
+            $dateInterval = new \DateInterval($data);
+        } catch (\Exception $e) {
+            throw new RuntimeException(sprintf('Invalid dateinterval "%s", expected ISO 8601 format', $data));
```
Comment (goetas) [inline src/JMS/Serializer/Handler/DateHandler.php]: what about adding the previous exception (`$e`) as third argument of `RuntimeException` ?

Comment (c0ntax) [inline src/JMS/Serializer/Handler/DateHandler.php]: I was just copying the one example of catching and rethrowing an exception within the project to keep things consistent, but I'm happy to add that as a 3rd param. It makes sense.

Comment (c0ntax) [inline src/JMS/Serializer/Handler/DateHandler.php]: Added rethrow of exception

**Comment by goetas** [comment]
@c0ntax thanks!

**Pull request merged**